### PR TITLE
community[minor]: Prem AI integration

### PR DESCRIPTION
**Pull request by Janaka-Steph**
This PR aims at adding [Prem AI](https://www.premai.io) support to LangChainJS through the Chat and Embeddings classes.
An account can be provided so that you can try running the tests.

The Prem team
https://twitter.com/premai_io

### CODE REVIEW

**Comment by jacoblee93** [comment]
Thank you! Will review as soon as I can.

**Comment by jacoblee93** [comment]
Opened #4790 with some build fixes, can't merge into this branch. Can you get those changes in here @Janaka-Steph?

Also, can you add some docs?

**Comment by Janaka-Steph** [comment]
> Opened #4790 with some build fixes, can't merge into this branch. Can you get those changes in here @Janaka-Steph?
> 
> Also, can you add some docs?

Thank you for the quick review. I made the requested changes. Let me know

File: libs/langchain-community/package.json [hunk 1]
```diff
@@ -489,6 +491,9 @@
     "portkey-ai": {
       "optional": true
     },
+    "premai": {
```
Comment (jacoblee93) [inline libs/langchain-community/package.json:494]: This should match `@premai/prem-sdk`

File: libs/langchain-community/src/chat_models/premai.ts [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+  CreateChatCompletionResponse,
+} from "@premai/prem-sdk";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+import {
+  ChatGeneration,
+  ChatGenerationChunk,
+  ChatResult,
+} from "@langchain/core/outputs";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { AxiosRequestConfig } from "axios";
```
Comment (jacoblee93) [inline libs/langchain-community/src/chat_models/premai.ts:32]: This is kinda incidental, can we make this a peer dep and install a version as part of setup?

Comment (jacoblee93) [inline libs/langchain-community/src/chat_models/premai.ts:32]: I assume this is an indirect dep via the Prem SDK?

File: libs/langchain-community/src/chat_models/premai.ts [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+  ): Promise<AsyncIterable<ChatCompletionStreamingCompletionData>>;
+
+  // @ts-expect-error - TS doesn't like the overloads
+  async completionWithRetry(
+    request: ChatCompletionCreateParamsNonStreaming,
+    options?: AxiosRequestConfig
+  ): Promise<CreateChatCompletionResponse>;
+
+  async completionWithRetry(
+    request: ChatCompletionCreateParamsStreaming,
```
Comment (jacoblee93) [inline libs/langchain-community/src/chat_models/premai.ts:297]: Pretty sure you can fix by making this:

```js
request: ChatCompletionCreateParamsStreaming | ChatCompletionCreateParamsNonStreaming
```

**Comment by jacoblee93** [comment]
Can you update deps and run `yarn` to update the lockfile?

Also, remember to run `yarn format` from root.

File: libs/langchain-community/src/chat_models/premai.ts (hunk 2 above)
Comment (Janaka-Steph) [reply libs/langchain-community/src/chat_models/premai.ts:32]: Prem SDK doesn't reexport AxiosRequestConfig. I didn't find a better solution than `any`. I think it's fine.

File: libs/langchain-community/package.json (hunk 1 above)
Comment (Janaka-Steph) [reply libs/langchain-community/package.json]: done

File: libs/langchain-community/src/chat_models/premai.ts (hunk 3 above)
Comment (Janaka-Steph) [reply libs/langchain-community/src/chat_models/premai.ts:296]: done

**Comment by Janaka-Steph** [comment]
Made the requested changes. Please have an other look. Thank you.

File: docs/core_docs/docs/integrations/chat/premai.mdx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import IntegrationInstallTooltip from "@mdx_components/integration_install_tooltip.mdx";
+
+<IntegrationInstallTooltip></IntegrationInstallTooltip>
+
+```bash npm2yarn
+npm install @langchain/community
+```
+
+import PremAI from "@examples/models/chat/integration_prem.ts";
```
Comment (jacoblee93) [inline docs/core_docs/docs/integrations/chat/premai.mdx:28]: Should be `integration_premai.ts`. Docs build is failing with:

```
Module not found: Error: Can't resolve '@examples/models/chat/integration_prem.ts' in '/vercel/path0/docs/core_docs/docs/integrations/chat'
```

File: libs/langchain-community/langchain.config.js [hunk 5]
```diff
@@ -155,6 +156,7 @@ export const config = {
     "chat_models/minimax": "chat_models/minimax",
     "chat_models/ollama": "chat_models/ollama",
     "chat_models/portkey": "chat_models/portkey",
+    "chat_models/premai": "chat_models/premai",
```
Comment (jacoblee93) [inline libs/langchain-community/langchain.config.js:159]: These also need to go under `requiresOptionalDependency`

**Comment by jacoblee93** [comment]
I think should be the last one - can also just push to your branch if you give me access

**Comment by jacoblee93** [comment]
Merged in #4867

**Comment by jacoblee93** [comment]
Thank you!

**Pull request closed without merging**